packages/legacy/core/App/components/modals/ErrorModal.tsx

```diff
@@ -68,8 +81,11 @@ const ErrorModal: React.FC<ErrorModalProps> = ({ reportProblemAction }) => {
           description={error ? error.description : t('Error.Problem')}
           message={formattedMessageForError(error ?? null)}
           onCallToActionPressed={onDismissModalTouched}
-          secondaryCallToActionTitle={t('Error.ReportThisProblem')}
-          secondaryCallToActionPressed={reportProblemAction && error ? () => reportProblemAction(error) : undefined}
+          secondaryCallToActionTitle={reported ? t('Error.Reported') : t('Error.ReportThisProblem')}
+          secondaryCallToActionDisabled={reported}
+          secondaryCallToActionIcon={reported ? <Icon style={{ marginRight: 8 }} name={'check-circle'} size={18} color={ColorPallet.semantic.success} /> : undefined}
```

Comment:
it would be nice to see this defined in a function and just pass the reported variable into it. I think it would make it easier to read

Reply:
done